### ISSUE TITLE
POSITION(in, search, startIndex) emulation returns wrong result when search was not found

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11295]
The current emulation of

```java
position(in, search, startIndex)
```

Is

```sql
(position(:search in substring(:in, :startIndex)) + (:startIndex - 1))
```

But that's correct only if we find `:in` in `:search`. If we don't the result should be `0`, but it is `startIndex`

**Comment by lukaseder** [comment]
It might be tempting to generate this:

```sql
coalesce(nullif(position(:search in substring(:in, :startIndex)), 0) + (:startIndex - 1), 0)
```

But that would be wrong as soon as any of the arguments is `null`, in case of which the result must be `null`, not `0`

**Comment by lukaseder** [comment]
`DECODE` will do (mostly emulated using `CASE`)

```sql
case position(:search in substring(:in, :startIndex))
  when 0 then 0
  else (position(:search in substring(:in, :startIndex)) + (:startIndex - 1))
end
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.6 (#11296)
